recipes/bison/all/conanfile.py

```diff
@@ -23,151 +25,149 @@ class BisonConan(ConanFile):
         "fPIC": True,
     }
 
-    exports_sources = "patches/*"
-
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
+    @property
+    def _user_info_build(self):
+        return getattr(self, "user_info_build", self.deps_user_info)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("m4/1.4.19")
 
     def validate(self):
-        if self.settings.compiler == "Visual Studio" and self.version == "3.8.2":
-            raise ConanInvalidConfiguration("bison/3.8.2 is not yet ready for Visual Studio, use previous version or open a pull request on https://github.com/conan-io/conan-center-index/pulls")
+        if is_msvc(self) and self.version == "3.8.2":
```

Comment:
I know this is meant to be just a port, but this might be a bug in the original recipe.
If someone ever bumps the version, it might break if the check is not updated. Maybe checking for anything `>=3.8.2` would be better until support for vs is implemented?

Reply:
AFAIR, it might be specific to 3.8.2, so the intend was to raise for this specific version with the hope of a fix from another contributor, but don't live with this situation forever. Therefore if someone submits a new version of bison, we may want to know whether it builds with Visual Studio instead of discarding this build silently.

Reply:
Oh then that makes lots of sense, thanks for the explanation!